### PR TITLE
Implemented op1Add

### PR DESCRIPTION
**Pull request by annaecamp**
* Implemented OP_1ADD
* Updated `writeNum` to `writeInt` because it was confusing to mix "Int" and "Num" for no good reason.

### CODE REVIEW

File: ops/ops.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func duplicate(v []byte) []byte {
 	return c
 }
 
-func writeNum(c Context, num int32) error {
+func writeInt(c Context, num int32) error {
```
Comment (benjic) [inline ops/ops.go]: 👏 This was confusing! Good work!

File: ops/stack.go [hunk 2]
```diff
@@ -82,13 +82,9 @@ func opDup(c Context) error {
 	}
 
 	v := c.Pop()
-	if v != nil {
-		v2 := make([]byte, len(v))
-		copy(v2, v)
-		c.Push(v2)
-	}
-
+	c.Push(duplicate(v))
```
Comment (benjic) [inline ops/stack.go]: 👏 Much cleaner!

File: ops/arithmetic.go [hunk 3]
```diff
@@ -32,7 +32,11 @@ const (
 )
 
 func op1Add(c Context) error {
-	// TODO: implement me
+	d, err := readInt(c)
+	if err != nil {
+		return err
+	}
+	writeInt(c, d+1)
```
Comment (benjic) [inline ops/arithmetic.go]: 👍 This will certainly add one to the top value of the stack.

File: ops/stack.go (hunk 2 above)
Comment (annaecamp) [reply ops/stack.go]: Thanks!

**Pull request merged**